pkg/kubelet/allocation/allocation_manager_test.go

```diff
@@ -177,8 +177,10 @@ func TestUpdatePodFromAllocation(t *testing.T) {
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
+			allocationManager := makeAllocationManager(t, &containertest.FakeRuntime{}, nil, nil)
 			pod := test.pod.DeepCopy()
-			allocatedPod, updated := updatePodFromAllocation(pod, test.allocated)
+			allocationManager.(*manager).allocated.SetPodResourceInfo(pod.UID, test.allocated)
```

Comment:
The type assertion to `*manager` to access the internal `allocated` field makes this test brittle. If the concrete type that implements the `allocationManager` interface changes in the future, this test will fail. Consider adding a test-only helper function to set this state to avoid breaking encapsulation.

Reply:
It doesn't seem worth exposing a test-only method for a unit test. I originally was going to call `SetAllocatedResources`, but this test is explicitly testing various weird allocated states to ensure the methods are robust to partial failures.